roles/testnode/tasks/apt_systems.yml

```diff
@@ -3,7 +3,7 @@
 # For anything ubuntu < 15.04 or debian <=7 we still do.
 - name: Setup local repo files.
   include: apt/repos.yml
-  when: ansible_distribution_major_version|int < 15
+  when: ansible_pkg_mgr == "apt" and ansible_distribution_major_version|int < 15
```

Comment:
@dmick So, add a `when:` with the same conditional to all the blocks in the file where it is currently missing?